### ISSUE TITLE
Project issue

### ISSUE THREAD

**Issue by konstantinosDMS** [kevinsmith/no-framework#4]
Hi,I am trying to make the project running,download dependencies via composer under www root folder of wamp but when I navigate to the public folder I get 404 not found.Any suggestion?

**Comment by kevinsmith** [comment]
Hmm, I'm not sure, but that may be something misconfigured with WAMP. What happens when you try just using PHP's built-in server?

```
php -S localhost:8080 -t path/to/public/
```

**Comment by konstantinosDMS** [comment]
The problem is with the return statement.I will take a closely look.

**Comment by kevinsmith** [comment]
Good deal, let me know what you find!

**Comment by konstantinosDMS** [comment]
When I am trying to run fast route I get this 
`C:\wamp64\www\no-framework-master\public\index.php:41:
object(FastRoute\Dispatcher\GroupCountBased)[25]
  protected 'staticRouteMap' => 
    array (size=1)
      'GET' => 
        array (size=1)
          '/hello' => string 'ExampleApp\HelloWorld' (length=21)
  protected 'variableRouteData' => 
    array (size=0)
      empty
about:blank
`
which seems ok. 

index.php:
```php
declare(strict_types=1);
use DI\ContainerBuilder;
use ExampleApp\HelloWorld;
use FastRoute\RouteCollector;
use Middlewares\FastRoute;
use Middlewares\RequestHandler;
use Relay\Relay;
use Zend\Diactoros\Response;
use Zend\Diactoros\Response\SapiEmitter;
use Zend\Diactoros\ServerRequestFactory;
use function DI\create;
use function DI\get;
use function FastRoute\simpleDispatcher;
require_once dirname(__DIR__) . '/vendor/autoload.php';
//$helloWorld = new \ExampleApp\HelloWorld();
//$helloWorld->announce();

$containerBuilder = new ContainerBuilder();
$containerBuilder->useAutowiring(false);
$containerBuilder->useAnnotations(false);
$containerBuilder->addDefinitions([
    HelloWorld::class => create(HelloWorld::class)
        ->constructor(get('Foo'), get('Response')),
    'Foo' => 'bar',
    'Response' => function() {
        return new Response();
    },
]);
// @noinspection PhpUnhandledExceptionInspection 
$container = $containerBuilder->build();
$helloWorld = $container->get(\ExampleApp\HelloWorld::class);
//$helloWorld->announce();
$routes = simpleDispatcher(function (RouteCollector $r) {
    $r->get('/hello', \ExampleApp\HelloWorld::class);
});
$middlewareQueue[] = new FastRoute($routes);
$middlewareQueue[] = new RequestHandler($container);
// @noinspection PhpUnhandledExceptionInspection 
$requestHandler = new Relay($middlewareQueue);
$response = $requestHandler->handle(ServerRequestFactory::fromGlobals());
/*
$emitter = new SapiEmitter();
// @noinspection PhpVoidFunctionResultUsedInspection
return $emitter->emit($response); 
 */
```

helloworld.php:
```
class HelloWorld
{
    /**
     * @var string
     */
    private $foo;
    /**
     * @var ResponseInterface
     */
    private $response;

    /**
     * HelloWorld constructor.
     *
     * @param ResponseInterface $response
     * @param string $foo
     */
    public function __construct(
        string $foo,
        ResponseInterface $response
    ) {
        $this->foo = $foo;
        $this->response = $response;
    }

    /**
     *
     *
     * @throws \RuntimeException
     * @throws \InvalidArgumentException
     */
   /* public function __invoke(): ResponseInterface
    {
        $response = $this->response->withHeader('Content-Type', 'text/html');
        $response->getBody()
            ->write("<html><head></head><body>Hello, {$this->foo} world!</body></html>");

        return $response;
    }*/
    
    /*public function announce(): void
    {
        echo 'Hello, autoloaded world!';
    }*/
    
    public function __invoke(): void
    {
        echo 'Hello, autoloaded world!';
        exit;
    }
}
```
I can;t get to http://no-framework:8090/hello
get 404 not found

Any suggestion?

**Comment by kevinsmith** [comment]
Having a little trouble following the code. Would you mind using [code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/)? (Even better with syntax highlighting!)

One thing I did see though: is your `HelloWorld` class's filename really `hellowrold.php` or is that a typo here in the issue? If it's misspelled, the autoloader won't be able to find it, so that could be your issue there.

**Comment by konstantinosDMS** [comment]
No it was a typo.The right is helloworld.php.Thanks

**Comment by kevinsmith** [comment]
Maybe it has something to do with your server setup? I only mention that because your URL is different than the tutorial (http://no-framework:8090/hello vs http://localhost:8080/hello). Using PHP's built-in web server, as shown in the tutorial, your code works for me.

<img width="652" alt="image" src="https://user-images.githubusercontent.com/397904/49329822-3a0fba00-f553-11e8-9e52-f54d720f99a8.png">

What happens when you try running

```
php -S localhost:8080 -t public/
```

from the command line?

**Comment by konstantinosDMS** [comment]
Thank you for your response. I fixed it and runs perfectly. You were right. I have installed on my PC xampp, wampp with deferent PHP versions and I had to change some things. But now runs perfectly. I am trying to write PHP based on PSR from now on and I have many questions so If you do not mind maybe I will ask you some things sometime. Right now I be prepared on ZEND Framework and everything is going well. My job is DBA for a ministry here in Greece and I have already work on Joomla and Wordpress, Dot Net Core. 

Thanks in advance and have a nice weekend!


Sent from Mail for Windows 10
